hera_stats/stats.py

```diff
@@ -6,157 +6,83 @@
-        groups = pc.spectra(jkstr)
-        ref = dict(zip([int(g[3:]) for g in groups], groups))
-        spec_l, err_l, grp_l = [],[],[]
-
-        for gi in sorted(ref.keys()):
-            g = ref[gi]
-
-            # Get delays, spectra, and errors
-            uvp = pc.get_pspec(jkstr,g)
-            dlys = uvp.get_dlys(0) * 10**9
-            key = uvp.get_all_keys()[0]
-            avspec = proj(uvp.get_data(key)[0])
-            errspec = proj(uvp.get_stats("bootstrap_errs", key)[0])
-            
-            spec_l.append(avspec)
-            err_l.append(errspec)
-            grp_l.append(uvp.labels)
-
-        # Sort by specific item if needed and pc has jackknife pairs
-        if isinstance(sortby, int) and len(spec_l) == 2:
-            # Check if item is in groups
-            ingrp = [str(sortby) in g for g in grp_l]
-
-            # If item not in either group, don't change anything
-            if all(np.equal(ingrp, False)):
-                if sortby is not None:
-                    nfail += 1
-                ingrp = [True, False]
-
-            # If sortby is in the second group, revese list
-            if all(np.equal(ingrp, [False, True])):
-                spec_l.reverse()
-                grp_l.reverse()
-                err_l.reverse()
-
-        # If reverse is requested, then reverse the lists
-        elif sortby == "reverse":
-            spec_l.reverse()
-            grp_l.reverse()
-            err_l.reverse()
-
-        # Calculate zscores for the list of spectra just loaded
-        z = standardize(spec_l, err_l, method=zscore)
-
-        spectra.append(spec_l)
-        errs.append(err_l)
-        grps.append(grp_l)
-        zs.append(z)
-
-    if nfail >= len(spectra):
-        raise ValueError("Sortby item %s not found" % str(sortby))
-
-    if sortby is None:
-        sortstring = ""
-    else:
-        sortstring = "(sorted by %s)" % str(sortby)
-
-    # Put everything into dictionary
-    dic = {"dlys": dlys, "spectra": np.array(spectra), "errs": np.array(errs),
-           "grps": grps, "jkftype": jkftype, "sortitem": sortby,
-           "sortstring": sortstring, "zscores": np.array(zs)}
+    jk = copy.deepcopy(jkset)
 
-    return dic
+    assert isinstance(jkset, jkset_lib.JKSet), "Expected jkset to be hera_stats.jkset.JKSet instance."
+    assert axis in [0, 1], "Axis must be either 1 or 0."
 
-def standardize(spectra, errs, method="weightedsum"):
+    # Do weighted sum calculation
+    aerrs = 1. / np.sum(jk.errs ** -2, axis=axis)
+    av = aerrs * np.sum(jk.spectra * jk.errs**-2, axis=axis)
+    std = (aerrs * len(jkset.spectra)) ** 0.5
+    
+    # Transpose jkset if necessary
+    if axis == 1:
+        jk = jk.T()
+
+    # Average or sum metadata
+    nsamp = np.sum(jk.nsamples, axis=0)
+    integrations = np.average(jk.integrations, axis=0)
+    times = np.average(jk.times, axis=0)
+
+    uvp_list = jk._uvp_list[0]
+
+    # Set UVPSpec attrs
+    for i, uvp in enumerate(uvp_list):
```

Comment:
Is this editing the data inside the `UVPSpec` objects, or working on a copy? It might cause weird behaviour if it's now working on a copy.